{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "27091086-bdcd-4b0e-80d8-10a66208460f",
   "metadata": {},
   "source": [
    "# Memote"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f2a81533-61b5-44c3-a722-5b4a146eab61",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "from cobra.io import read_sbml_model, write_sbml_model\n",
    "from cobra import Reaction, Metabolite"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a384de8d-c868-4b74-8129-9b121b6bbc89",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "from cobra.io import read_sbml_model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3ef78d05-214b-49ba-8d46-35ee8a7250ab",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Usage: memote [OPTIONS] COMMAND [ARGS]...\n",
      "\n",
      "  Metabolic model testing command line tool.\n",
      "\n",
      "  In its basic invocation memote runs a test suite on a metabolic model.\n",
      "  Through various subcommands it can further generate three types of pretty\n",
      "  HTML reports (snapshot, diff and history), generate a model repository\n",
      "  structure for starting a new project, and recreate the test result history.\n",
      "\n",
      "Options:\n",
      "  -h, --help           Show this message and exit.\n",
      "  -V, --version        Show the version and exit.\n",
      "  -v, --verbosity LVL  Either CRITICAL, ERROR, WARNING, INFO or DEBUG\n",
      "                       [default: INFO]\n",
      "\n",
      "Commands:\n",
      "  history  Re-compute test results for the git branch history.\n",
      "  new      Create a suitable model repository structure from a template.\n",
      "  online   Upload the repository to GitHub and create a gh-pages branch.\n",
      "  report   Generate one of three different types of reports.\n",
      "  run      Run the test suite on a single model and collect results.\n"
     ]
    }
   ],
   "source": [
    "!memote "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f1fa229f-76e4-4c70-a049-772af8fdb840",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No objective coefficients in model. Unclear what should be optimized\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCW773</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>157248ed0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>949</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1207</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of genes</strong></td>\n",
       "                <td>773</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td></td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCW773 at 0x157248ed0>"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCW773 =  read_sbml_model('/Users/macintosh/Downloads/iCW773.xml')\n",
    "\n",
    "iCW773"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "bdfe59ae-2b2c-4643-8f48-39dc0496e716",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "SBML package 'layout' not supported by cobrapy, information is not parsed\n",
      "https://juser.fz-juelich.de/record/188973 does not conform to 'http(s)://identifiers.org/collection/id' or'http(s)://identifiers.org/COLLECTION:id\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCBG21FR</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>16b919c90</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1042</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1539</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of genes</strong></td>\n",
       "                <td>805</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>102</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*Growth - 1.0*Growth_reverse_699ae</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, periplasm, extracellular space</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCBG21FR at 0x16b919c90>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCGB21FR =  read_sbml_model('/Users/macintosh/Downloads/iCGB21FR.xml')\n",
    "\n",
    "iCGB21FR"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "084b0e66-f30a-487a-b844-576c7d0bafa7",
   "metadata": {},
   "source": [
    "## iCW773"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "533e5c1f-66bf-446e-945b-fc2e5c8b9d91",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "\u001b[31merror: \u001b[0mNo objective coefficients in model. Unclear what should be optimized\n",
      "\u001b[1m============================= test session starts ==============================\u001b[0m\n",
      "platform darwin -- Python 3.11.5, pytest-7.4.0, pluggy-1.0.0\n",
      "rootdir: /Users/macintosh\n",
      "plugins: anyio-3.5.0\n",
      "collected 146 items / 1 skipped                                                \u001b[0m\n",
      "\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py \u001b[31mF\u001b[0m\u001b[31m [  0%]\n",
      "\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31m         [ 44%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py \u001b[32m.\u001b[0m\u001b[31m [ 45%]\n",
      "\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31m                                                   [ 60%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py \u001b[32m.\u001b[0m\u001b[31m [ 60%]\n",
      "\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31m                                                                [ 67%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py \u001b[31mF\u001b[0m\u001b[31m [ 67%]\n",
      "\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31m                                             [ 86%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_essentiality.py \u001b[33ms\u001b[0m\u001b[31m [ 87%]\n",
      "\u001b[0m\u001b[31m                                                                         [ 87%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_growth.py \u001b[33ms\u001b[0m\u001b[31m [ 88%]\n",
      "\u001b[0m\u001b[31m                                                                         [ 88%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_matrix.py \u001b[32m.\u001b[0m\u001b[31m [ 89%]\n",
      "\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31m                                                                      [ 91%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbml.py \u001b[32m.\u001b[0m\u001b[31m [ 91%]\n",
      "\u001b[0m\u001b[32m.\u001b[0m\u001b[31m                                                                        [ 92%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py \u001b[32m.\u001b[0m\u001b[31m [ 93%]\n",
      "\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[33ms\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[33ms\u001b[0m\u001b[33ms\u001b[0m\u001b[31mF\u001b[0m\u001b[31m                                                               [100%]\u001b[0m\n",
      "\n",
      "\u001b[33m=============================== warnings summary ===============================\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/_pytest/config/__init__.py:1204\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/_pytest/config/__init__.py:1204: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: anyio\n",
      "    self._mark_plugins_for_rewrite(hook)\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::test_biomass_consistency[CG_biomass_cgl_ATCC13032]\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 26.227799999999995 is not an integer (in formula C26.227799999999995H47.309599999999996O21.0818)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::test_biomass_consistency[CG_biomass_cgl_ATCC13032]\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 47.309599999999996 is not an integer (in formula C26.227799999999995H47.309599999999996O21.0818)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::test_biomass_consistency[CG_biomass_cgl_ATCC13032]\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 21.0818 is not an integer (in formula C26.227799999999995H47.309599999999996O21.0818)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::test_biomass_consistency[CG_biomass_cgl_ATCC13032]\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:191: UserWarning: The element 'R' does not appear in the periodic table\n",
      "    warn(f\"The element {e} does not appear in the periodic table\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 29.018 is not an integer (in formula C29.018H52.94199999999999O15P)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 52.94199999999999 is not an integer (in formula C29.018H52.94199999999999O15P)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 41.018 is not an integer (in formula C41.018H74.942O29P2)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 74.942 is not an integer (in formula C41.018H74.942O29P2)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 17.018 is not an integer (in formula C17.018H30.941999999999997O1)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 30.941999999999997 is not an integer (in formula C17.018H30.941999999999997O1)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 64.598238154 is not an integer (in formula C64.598238154H117.50071172599999O7.357917P0.267926)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 117.50071172599999 is not an integer (in formula C64.598238154H117.50071172599999O7.357917P0.267926)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 7.357917 is not an integer (in formula C64.598238154H117.50071172599999O7.357917P0.267926)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.267926 is not an integer (in formula C64.598238154H117.50071172599999O7.357917P0.267926)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 19.222443 is not an integer (in formula C19.222443H38.476530999999966N5.458296999999999O10.650917999999953P-0.000000000000004S0.108442)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 38.476530999999966 is not an integer (in formula C19.222443H38.476530999999966N5.458296999999999O10.650917999999953P-0.000000000000004S0.108442)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 5.458296999999999 is not an integer (in formula C19.222443H38.476530999999966N5.458296999999999O10.650917999999953P-0.000000000000004S0.108442)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 10.650917999999953 is not an integer (in formula C19.222443H38.476530999999966N5.458296999999999O10.650917999999953P-0.000000000000004S0.108442)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.108442 is not an integer (in formula C19.222443H38.476530999999966N5.458296999999999O10.650917999999953P-0.000000000000004S0.108442)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 31.685756 is not an integer (in formula C31.685756H40.884037000000006N9.559588O20.989518P3.134647S0.355614)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 40.884037000000006 is not an integer (in formula C31.685756H40.884037000000006N9.559588O20.989518P3.134647S0.355614)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 9.559588 is not an integer (in formula C31.685756H40.884037000000006N9.559588O20.989518P3.134647S0.355614)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 20.989518 is not an integer (in formula C31.685756H40.884037000000006N9.559588O20.989518P3.134647S0.355614)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 3.134647 is not an integer (in formula C31.685756H40.884037000000006N9.559588O20.989518P3.134647S0.355614)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.355614 is not an integer (in formula C31.685756H40.884037000000006N9.559588O20.989518P3.134647S0.355614)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 1.809384 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.452346 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 1.514453 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.542815 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.002624 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.073913 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 20.35556 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.904692 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.072104 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.033655 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.035554 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 46.2 is not an integer (in formula C46.2H80.208N3O15P2)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 80.208 is not an integer (in formula C46.2H80.208N3O15P2)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 20.1 is not an integer (in formula C20.1H38.104O7P)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 38.104 is not an integer (in formula C20.1H38.104O7P)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 40.2 is not an integer (in formula C40.2H75.208O13P2)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 75.208 is not an integer (in formula C40.2H75.208O13P2)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 52.881848160000004 is not an integer (in formula C52.881848160000004H98.21897442240001N0.326586O16.9293528P2.563819800000001)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 98.21897442240001 is not an integer (in formula C52.881848160000004H98.21897442240001N0.326586O16.9293528P2.563819800000001)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.326586 is not an integer (in formula C52.881848160000004H98.21897442240001N0.326586O16.9293528P2.563819800000001)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 16.9293528 is not an integer (in formula C52.881848160000004H98.21897442240001N0.326586O16.9293528P2.563819800000001)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 2.563819800000001 is not an integer (in formula C52.881848160000004H98.21897442240001N0.326586O16.9293528P2.563819800000001)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 28.60628 is not an integer (in formula C28.60628H33.27465N11.5971O22.976729999999996P2.99579)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 33.27465 is not an integer (in formula C28.60628H33.27465N11.5971O22.976729999999996P2.99579)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 11.5971 is not an integer (in formula C28.60628H33.27465N11.5971O22.976729999999996P2.99579)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 22.976729999999996 is not an integer (in formula C28.60628H33.27465N11.5971O22.976729999999996P2.99579)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 2.99579 is not an integer (in formula C28.60628H33.27465N11.5971O22.976729999999996P2.99579)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 64.087066 is not an integer (in formula C64.087066H109.98235400000002N13.276022O50.168144P3.2394)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 109.98235400000002 is not an integer (in formula C64.087066H109.98235400000002N13.276022O50.168144P3.2394)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 13.276022 is not an integer (in formula C64.087066H109.98235400000002N13.276022O50.168144P3.2394)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 50.168144 is not an integer (in formula C64.087066H109.98235400000002N13.276022O50.168144P3.2394)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 3.2394 is not an integer (in formula C64.087066H109.98235400000002N13.276022O50.168144P3.2394)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html\n",
      "\u001b[36m\u001b[1m=========================== short test summary info ============================\u001b[0m\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_presence\u001b[0m - AssertionError: A total of 949 metabolites (100.00%) lack any form of annot...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_presence\u001b[0m - AssertionError: A total of 1207 reactions (100.00%) lack any form of annota...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_presence\u001b[0m - AssertionError: A total of 773 genes (100.00%) lack any form of         ann...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[pubchem.compound]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[kegg.compound]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[seed.compound]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[inchikey]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[inchi]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[chebi]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[hmdb]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[reactome]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[metanetx.chemical]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[bigg.metabolite]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[biocyc]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[rhea]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[kegg.reaction]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[seed.reaction]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[metanetx.reaction]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[bigg.reaction]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[reactome]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[ec-code]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[brenda]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[biocyc]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[refseq]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for refseq:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[uniprot]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for uniprot:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ecogene]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for ecogene:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[kegg.genes]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for kegg....\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ncbigi]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for ncbigi:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ncbigene]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for ncbig...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ncbiprotein]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for ncbip...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ccds]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for ccds:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[hprd]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for hprd:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[asap]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for asap:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[pubchem.compound]\u001b[0m - AssertionError: There are no metabolite annotations for the pubchem.compoun...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[kegg.compound]\u001b[0m - AssertionError: There are no metabolite annotations for the kegg.compound d...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[seed.compound]\u001b[0m - AssertionError: There are no metabolite annotations for the seed.compound d...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[inchikey]\u001b[0m - AssertionError: There are no metabolite annotations for the inchikey database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[inchi]\u001b[0m - AssertionError: There are no metabolite annotations for the inchi database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[chebi]\u001b[0m - AssertionError: There are no metabolite annotations for the chebi database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[hmdb]\u001b[0m - AssertionError: There are no metabolite annotations for the hmdb database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[reactome]\u001b[0m - AssertionError: There are no metabolite annotations for the reactome database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[metanetx.chemical]\u001b[0m - AssertionError: There are no metabolite annotations for the metanetx.chemical\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[bigg.metabolite]\u001b[0m - AssertionError: There are no metabolite annotations for the bigg.metabolite...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[biocyc]\u001b[0m - AssertionError: There are no metabolite annotations for the biocyc database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[rhea]\u001b[0m - AssertionError: There are no reaction annotations for the rhea database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[kegg.reaction]\u001b[0m - AssertionError: There are no reaction annotations for the kegg.reaction dat...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[seed.reaction]\u001b[0m - AssertionError: There are no reaction annotations for the seed.reaction dat...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[metanetx.reaction]\u001b[0m - AssertionError: There are no reaction annotations for the metanetx.reaction...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[bigg.reaction]\u001b[0m - AssertionError: There are no reaction annotations for the bigg.reaction dat...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[reactome]\u001b[0m - AssertionError: There are no reaction annotations for the reactome database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[ec-code]\u001b[0m - AssertionError: There are no reaction annotations for the ec-code database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[brenda]\u001b[0m - AssertionError: There are no reaction annotations for the brenda database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[biocyc]\u001b[0m - AssertionError: There are no reaction annotations for the biocyc database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[refseq]\u001b[0m - AssertionError: There are no gene annotations for the refseq database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[uniprot]\u001b[0m - AssertionError: There are no gene annotations for the uniprot database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ecogene]\u001b[0m - AssertionError: There are no gene annotations for the ecogene database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[kegg.genes]\u001b[0m - AssertionError: There are no gene annotations for the kegg.genes database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ncbigi]\u001b[0m - AssertionError: There are no gene annotations for the ncbigi database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ncbigene]\u001b[0m - AssertionError: There are no gene annotations for the ncbigene database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ncbiprotein]\u001b[0m - AssertionError: There are no gene annotations for the ncbiprotein database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ccds]\u001b[0m - AssertionError: There are no gene annotations for the ccds database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[hprd]\u001b[0m - AssertionError: There are no gene annotations for the hprd database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[asap]\u001b[0m - AssertionError: There are no gene annotations for the asap database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_gene_protein_reaction_rule_presence\u001b[0m - AssertionError: There are a total of 247 reactions (20.46%) without GPR:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_ngam_presence\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_compartments_presence\u001b[0m - AssertionError: A total of 1 compartments are defined in the model:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_protein_complex_presence\u001b[0m -   File \"<unknown>\", line 1\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_transport_reactions\u001b[0m - AssertionError: A total of 0 (0.00%) transport reactions are defined in the\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_constrained_transport_reactions\u001b[0m - ZeroDivisionError: division by zero\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_transport_reaction_gpr_presence\u001b[0m - ZeroDivisionError: division by zero\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_unique_metabolites\u001b[0m - AssertionError: Not counting the same entities in other compartments, there...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_reactions_with_identical_genes\u001b[0m - AssertionError: Based only on equal genes there are 102 different groups of\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_medium_metabolites\u001b[0m - TypeError: unsupported operand type(s) for /: 'int' and 'list'\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_biomass_consistency[CG_biomass_cgl_ATCC13032]\u001b[0m - TypeError: '<' not supported between instances of 'float' and 'NoneType'\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_biomass_precursors_default_production[CG_biomass_cgl_ATCC13032]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_biomass_precursors_open_production[CG_biomass_cgl_ATCC13032]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_gam_in_biomass[CG_biomass_cgl_ATCC13032]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_fast_growth_default[CG_biomass_cgl_ATCC13032]\u001b[0m - AssertionError: Using the biomass reaction CG_biomass_cgl_ATCC13032 and whe...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_direct_metabolites_in_biomass[CG_biomass_cgl_ATCC13032]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_essential_precursors_not_in_biomass[CG_biomass_cgl_ATCC13032]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_stoichiometric_consistency\u001b[0m - AssertionError: This model's stoichiometry inconsistent\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_unconserved_metabolites\u001b[0m - AssertionError: This model contains 5 unconserved metabolites: h_c, ppt_c, ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_inconsistent_min_stoichiometry\u001b[0m - TypeError: sequence item 0: expected str instance, list found\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM3]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM63]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM51]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM121]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM423]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM6]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM10]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM38]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM208]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM191]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM223]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM7517]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM12233]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM558]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM21]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM89557]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_reaction_mass_balance\u001b[0m - AssertionError: A total of 32 (3.07%) reactions are mass unbalanced with at...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_blocked_reactions\u001b[0m - AssertionError: There are 439 (36.37%) blocked reactions in         the mod...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_stoichiometrically_balanced_cycles\u001b[0m - AssertionError: There are 15 (1.24%) reactions         which participate in...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_deadends\u001b[0m - AssertionError: A total of 0 (0.00%) metabolites are not consumed by any re...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_metabolites_not_produced_with_open_bounds\u001b[0m - AssertionError: A total of 464 (48.89%) metabolites cannot be produced in c...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_metabolites_not_consumed_with_open_bounds\u001b[0m - AssertionError: A total of 464 (48.89%) metabolites cannot be consumed in c...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_reactions_unbounded_flux_default_condition\u001b[0m - AssertionError:  A fraction of 55.60% of the non-blocked reactions (in tota...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_biomass_specific_sbo_presence\u001b[0m - AssertionError: A total of 1 biomass reactions (100.00% of all biomass reac...\n",
      "\u001b[31m====== \u001b[31m\u001b[1m107 failed\u001b[0m, \u001b[32m34 passed\u001b[0m, \u001b[33m6 skipped\u001b[0m, \u001b[33m61 warnings\u001b[0m\u001b[31m in 172.19s (0:02:52)\u001b[0m\u001b[31m ======\u001b[0m\n",
      "Writing snapshot report to '/Users/macintosh/Downloads/iCW773.html'.\n"
     ]
    }
   ],
   "source": [
    "!memote report snapshot /Users/macintosh/Downloads/iCW773.xml --filename /Users/macintosh/Downloads/iCW773.html"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dadada83-5efd-457c-9f01-10f827d8f9c7",
   "metadata": {},
   "source": [
    "## iCGB21FR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4b621e99-619b-46dd-9cef-e74d264a6bdb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "\u001b[33mwarning: \u001b[0mSBML package 'layout' not supported by cobrapy, information is not parsed\n",
      "\u001b[33mwarning: \u001b[0mhttps://juser.fz-juelich.de/record/188973 does not conform to 'http(s)://identifiers.org/collection/id' or'http(s)://identifiers.org/COLLECTION:id\n",
      "\u001b[1m============================= test session starts ==============================\u001b[0m\n",
      "platform darwin -- Python 3.11.5, pytest-7.4.0, pluggy-1.0.0\n",
      "rootdir: /Users/macintosh\n",
      "plugins: anyio-3.5.0\n",
      "collected 146 items / 1 skipped                                                \u001b[0m\n",
      "\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py \u001b[31mF\u001b[0m\u001b[31m [  0%]\n",
      "\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31m         [ 44%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py \u001b[32m.\u001b[0m\u001b[31m [ 45%]\n",
      "\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31m                                                   [ 60%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py \u001b[32m.\u001b[0m\u001b[31m [ 60%]\n",
      "\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31m                                                                [ 67%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py \u001b[31mF\u001b[0m\u001b[31m [ 67%]\n",
      "\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[33ms\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[33ms\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31m                                             [ 86%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_essentiality.py \u001b[33ms\u001b[0m\u001b[31m [ 87%]\n",
      "\u001b[0m\u001b[31m                                                                         [ 87%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_growth.py \u001b[33ms\u001b[0m\u001b[31m [ 88%]\n",
      "\u001b[0m\u001b[31m                                                                         [ 88%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_matrix.py \u001b[32m.\u001b[0m\u001b[31m [ 89%]\n",
      "\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31m                                                                      [ 91%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbml.py \u001b[32m.\u001b[0m\u001b[31m [ 91%]\n",
      "\u001b[0m\u001b[32m.\u001b[0m\u001b[31m                                                                        [ 92%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py \u001b[31mF\u001b[0m\u001b[31m [ 93%]\n",
      "\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[33ms\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31m                                                               [100%]\u001b[0m\n",
      "\n",
      "\u001b[33m=============================== warnings summary ===============================\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/_pytest/config/__init__.py:1204\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/_pytest/config/__init__.py:1204: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: anyio\n",
      "    self._mark_plugins_for_rewrite(hook)\n",
      "\n",
      "-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html\n",
      "\u001b[36m\u001b[1m=========================== short test summary info ============================\u001b[0m\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_presence\u001b[0m - AssertionError: A total of 1 metabolites (0.10%) lack any form of annotation:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_presence\u001b[0m - AssertionError: A total of 3 reactions (0.19%) lack any form of annotation:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_presence\u001b[0m - AssertionError: A total of 24 genes (2.98%) lack any form of         annota...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[pubchem.compound]\u001b[0m - AssertionError: The following 1042 metabolites (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[kegg.compound]\u001b[0m - AssertionError: The following 234 metabolites (22.46%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[seed.compound]\u001b[0m - AssertionError: The following 77 metabolites (7.39%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[inchikey]\u001b[0m - AssertionError: The following 187 metabolites (17.95%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[inchi]\u001b[0m - AssertionError: The following 1042 metabolites (100.00%) lack annotation fo...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[chebi]\u001b[0m - AssertionError: The following 206 metabolites (19.77%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[hmdb]\u001b[0m - AssertionError: The following 326 metabolites (31.29%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[reactome]\u001b[0m - AssertionError: The following 575 metabolites (55.18%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[metanetx.chemical]\u001b[0m - AssertionError: The following 3 metabolites (0.29%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[bigg.metabolite]\u001b[0m - AssertionError: The following 2 metabolites (0.19%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[biocyc]\u001b[0m - AssertionError: The following 201 metabolites (19.29%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[rhea]\u001b[0m - AssertionError: The following 858 reactions (55.75%) lack annotation for rhea:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[kegg.reaction]\u001b[0m - AssertionError: The following 1044 reactions (67.84%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[seed.reaction]\u001b[0m - AssertionError: The following 574 reactions (37.30%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[metanetx.reaction]\u001b[0m - AssertionError: The following 223 reactions (14.49%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[bigg.reaction]\u001b[0m - AssertionError: The following 36 reactions (2.34%) lack annotation for bigg...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[reactome]\u001b[0m - AssertionError: The following 1355 reactions (88.04%) lack annotation for r...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[ec-code]\u001b[0m - AssertionError: The following 831 reactions (54.00%) lack annotation for ec...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[brenda]\u001b[0m - AssertionError: The following 1539 reactions (100.00%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[biocyc]\u001b[0m - AssertionError: The following 898 reactions (58.35%) lack annotation for bi...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[refseq]\u001b[0m - AssertionError: The following 805 genes (100.00%) lack annotation for refseq:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[uniprot]\u001b[0m - AssertionError: The following 805 genes (100.00%) lack annotation for uniprot:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ecogene]\u001b[0m - AssertionError: The following 805 genes (100.00%) lack annotation for ecogene:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[kegg.genes]\u001b[0m - AssertionError: The following 68 genes (8.45%) lack annotation for kegg.genes:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ncbigi]\u001b[0m - AssertionError: The following 805 genes (100.00%) lack annotation for ncbigi:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ncbigene]\u001b[0m - AssertionError: The following 68 genes (8.45%) lack annotation for ncbigene:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ncbiprotein]\u001b[0m - AssertionError: The following 24 genes (2.98%) lack annotation for ncbiprot...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ccds]\u001b[0m - AssertionError: The following 805 genes (100.00%) lack annotation for ccds:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[hprd]\u001b[0m - AssertionError: The following 805 genes (100.00%) lack annotation for hprd:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[asap]\u001b[0m - AssertionError: The following 805 genes (100.00%) lack annotation for asap:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[pubchem.compound]\u001b[0m - AssertionError: There are no metabolite annotations for the pubchem.compoun...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[kegg.compound]\u001b[0m - AssertionError: A total of 274 metabolite annotations (33.91%) do not match...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[seed.compound]\u001b[0m - AssertionError: A total of 40 metabolite annotations (4.15%) do not match the\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[inchi]\u001b[0m - AssertionError: There are no metabolite annotations for the inchi database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[hmdb]\u001b[0m - AssertionError: A total of 181 metabolite annotations (25.28%) do not match...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[reactome]\u001b[0m - AssertionError: A total of 411 metabolite annotations (88.01%) do not match...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[ec-code]\u001b[0m - AssertionError: A total of 4 reaction annotations (0.26%) do not match the\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[brenda]\u001b[0m - AssertionError: There are no reaction annotations for the brenda database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[refseq]\u001b[0m - AssertionError: There are no gene annotations for the refseq database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[uniprot]\u001b[0m - AssertionError: There are no gene annotations for the uniprot database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ecogene]\u001b[0m - AssertionError: There are no gene annotations for the ecogene database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ncbigi]\u001b[0m - AssertionError: There are no gene annotations for the ncbigi database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ncbigene]\u001b[0m - AssertionError: A total of 737 gene annotations (91.55%) do not match the\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ncbiprotein]\u001b[0m - AssertionError: A total of 1 gene annotations (0.12%) do not match the     ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ccds]\u001b[0m - AssertionError: There are no gene annotations for the ccds database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[hprd]\u001b[0m - AssertionError: There are no gene annotations for the hprd database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[asap]\u001b[0m - AssertionError: There are no gene annotations for the asap database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_gene_protein_reaction_rule_presence\u001b[0m - AssertionError: There are a total of 276 reactions (17.93%) without GPR:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_transport_reaction_gpr_presence\u001b[0m - AssertionError: There are a total of 110 transport reactions (24.72% of all\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_duplicate_metabolites_in_compartments\u001b[0m - TypeError: sequence item 0: expected str instance, tuple found\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_reactions_with_partially_identical_annotations\u001b[0m - AssertionError: Based on annotations there are 97 different groups of overl...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_duplicate_reactions\u001b[0m - TypeError: sequence item 0: expected str instance, tuple found\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_reactions_with_identical_genes\u001b[0m - AssertionError: Based only on equal genes there are 172 different groups of\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_medium_metabolites\u001b[0m - TypeError: unsupported operand type(s) for /: 'int' and 'list'\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_biomass_consistency[Growth]\u001b[0m - AssertionError: The component molar mass of the biomass reaction Growth sum...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_gam_in_biomass[Growth]\u001b[0m - AssertionError: No, Growth does not contain a term for growth-associated\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_fast_growth_default[Growth]\u001b[0m - AssertionError: Using the biomass reaction Growth and when the model is sim...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_essential_precursors_not_in_biomass[Growth]\u001b[0m - AssertionError: Growth lacks a total of 1 essential metabolites         (1....\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_stoichiometric_consistency\u001b[0m - AssertionError: This model's stoichiometry inconsistent\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_unconserved_metabolites\u001b[0m - AssertionError: This model contains 3 unconserved metabolites: h_c, h_p, h_e\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_inconsistent_min_stoichiometry\u001b[0m - TypeError: sequence item 0: expected str instance, list found\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_reaction_charge_balance\u001b[0m - AssertionError: A total of 28 (2.07%) reactions are charge unbalanced with at\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_reaction_mass_balance\u001b[0m - AssertionError: A total of 11 (0.81%) reactions are mass unbalanced with at...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_blocked_reactions\u001b[0m - AssertionError: There are 48 (3.12%) blocked reactions in         the model:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_stoichiometrically_balanced_cycles\u001b[0m - AssertionError: There are 126 (8.19%) reactions         which participate i...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_orphans\u001b[0m - AssertionError: A total of 1 (0.10%) metabolites are not produced by any re...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_deadends\u001b[0m - AssertionError: A total of 1 (0.10%) metabolites are not consumed by any re...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_metabolites_not_produced_with_open_bounds\u001b[0m - AssertionError: A total of 151 (14.49%) metabolites cannot be produced in c...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_metabolites_not_consumed_with_open_bounds\u001b[0m - AssertionError: A total of 138 (13.24%) metabolites cannot be consumed in c...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_reactions_unbounded_flux_default_condition\u001b[0m - AssertionError:  A fraction of 15.79% of the non-blocked reactions (in tota...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_metabolite_sbo_presence\u001b[0m - AssertionError: A total of 16 metabolites (1.54%) lack annotation with any ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_reaction_sbo_presence\u001b[0m - AssertionError: A total of 3 reactions (0.19%) lack annotation with any type\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_gene_sbo_presence\u001b[0m - AssertionError: A total of 25 genes (3.11%) lack annotation with any type of\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_metabolic_reaction_specific_sbo_presence\u001b[0m - AssertionError: A total of 604 metabolic reactions (66.59% of all purely\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_transport_reaction_specific_sbo_presence\u001b[0m - AssertionError: A total of 9 metabolic reactions (2.02% of all transport\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_metabolite_specific_sbo_presence\u001b[0m - AssertionError: A total of 16 transport reactions (1.54% of all metabolites)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_gene_specific_sbo_presence\u001b[0m - AssertionError: A total of 25 genes (3.11% of all genes) lack             a...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_exchange_specific_sbo_presence\u001b[0m - AssertionError: A total of 1 exchange reactions (0.55% of all exchange\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_sink_specific_sbo_presence\u001b[0m - AssertionError: A total of 0 genes (0.00% of all sink reactions) lack\n",
      "\u001b[31m======= \u001b[31m\u001b[1m82 failed\u001b[0m, \u001b[32m59 passed\u001b[0m, \u001b[33m6 skipped\u001b[0m, \u001b[33m1 warning\u001b[0m\u001b[31m in 210.31s (0:03:30)\u001b[0m\u001b[31m ========\u001b[0m\n",
      "Writing snapshot report to '/Users/macintosh/Downloads/iCGB21FR.html'.\n"
     ]
    }
   ],
   "source": [
    "!memote report snapshot /Users/macintosh/Downloads/iCGB21FR.xml --filename /Users/macintosh/Downloads/iCGB21FR.html"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4f1f138d-1c16-41f3-93d5-2da75b4d66cd",
   "metadata": {},
   "source": [
    "## Compare the performance of the two models"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "d56a4ef8-a9f0-4f31-974b-d65fdc77d625",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "\u001b[33mwarning: \u001b[0mSBML package 'layout' not supported by cobrapy, information is not parsed\n",
      "\u001b[33mwarning: \u001b[0mhttps://juser.fz-juelich.de/record/188973 does not conform to 'http(s)://identifiers.org/collection/id' or'http(s)://identifiers.org/COLLECTION:id\n",
      "The current solver interface glpk doesn't support setting the optimality tolerance.\n",
      "\u001b[31merror: \u001b[0mNo objective coefficients in model. Unclear what should be optimized\n",
      "\u001b[1m============================= test session starts ==============================\u001b[0m\n",
      "platform darwin -- Python 3.11.5, pytest-7.4.0, pluggy-1.0.0\n",
      "rootdir: /Users/macintosh\n",
      "plugins: anyio-3.5.0\n",
      "\u001b[1mcollecting ... \u001b[0m\u001b[1m============================= test session starts ==============================\u001b[0m\n",
      "platform darwin -- Python 3.11.5, pytest-7.4.0, pluggy-1.0.0\n",
      "rootdir: /Users/macintosh\n",
      "plugins: anyio-3.5.0\n",
      "collected 146 items / 1 skipped                                                \u001b[0m\n",
      "\n",
      "collected 146 items / 1 skipped                                                \u001b[0mpy \u001b[1m\n",
      "\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py \u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31m [  0%]\n",
      "\u001b[0m\u001b[31m [  0%]\n",
      "\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31m         [ 44%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py \u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31m [ 45%]\n",
      "\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31m         [ 44%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py \u001b[32m.\u001b[0m\u001b[31m [ 45%]\n",
      "\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31m                                                   [ 60%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py \u001b[32m.\u001b[0m\u001b[31m [ 60%]\n",
      "\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31m                                                                [ 67%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py \u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31m                                                   [ 60%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py \u001b[32m.\u001b[0m\u001b[31m [ 60%]\n",
      "\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31m [ 67%]\n",
      "\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31m                                                                [ 67%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py \u001b[31mF\u001b[0m\u001b[31m [ 67%]\n",
      "\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[33ms\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[33ms\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31m                                             [ 86%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_essentiality.py \u001b[33ms\u001b[0m\u001b[31m [ 87%]\n",
      "\u001b[0m\u001b[31m                                                                         [ 87%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_growth.py \u001b[33ms\u001b[0m\u001b[31m [ 88%]\n",
      "\u001b[0m\u001b[31m                                                                         [ 88%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_matrix.py \u001b[32m.\u001b[0m\u001b[31m [ 89%]\n",
      "\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31m                                                                      [ 91%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbml.py \u001b[32m.\u001b[0m\u001b[31m [ 91%]\n",
      "\u001b[0m\u001b[32m.\u001b[0m\u001b[31m                                                                        [ 92%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py \u001b[31mF\u001b[0m\u001b[31m [ 93%]\n",
      "\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[33ms\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31m                                                               [100%]\u001b[0m\n",
      "\n",
      "\u001b[33m=============================== warnings summary ===============================\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/_pytest/config/__init__.py:1204\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/_pytest/config/__init__.py:1204: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: anyio\n",
      "    self._mark_plugins_for_rewrite(hook)\n",
      "\n",
      "-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html\n",
      "\u001b[36m\u001b[1m=========================== short test summary info ============================\u001b[0m\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_presence\u001b[0m - AssertionError: A total of 1 metabolites (0.10%) lack any form of annotation:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_presence\u001b[0m - AssertionError: A total of 3 reactions (0.19%) lack any form of annotation:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_presence\u001b[0m - AssertionError: A total of 24 genes (2.98%) lack any form of         annota...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[pubchem.compound]\u001b[0m - AssertionError: The following 1042 metabolites (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[kegg.compound]\u001b[0m - AssertionError: The following 234 metabolites (22.46%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[seed.compound]\u001b[0m - AssertionError: The following 77 metabolites (7.39%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[inchikey]\u001b[0m - AssertionError: The following 187 metabolites (17.95%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[inchi]\u001b[0m - AssertionError: The following 1042 metabolites (100.00%) lack annotation fo...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[chebi]\u001b[0m - AssertionError: The following 206 metabolites (19.77%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[hmdb]\u001b[0m - AssertionError: The following 326 metabolites (31.29%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[reactome]\u001b[0m - AssertionError: The following 575 metabolites (55.18%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[metanetx.chemical]\u001b[0m - AssertionError: The following 3 metabolites (0.29%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[bigg.metabolite]\u001b[0m - AssertionError: The following 2 metabolites (0.19%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[biocyc]\u001b[0m - AssertionError: The following 201 metabolites (19.29%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[rhea]\u001b[0m - AssertionError: The following 858 reactions (55.75%) lack annotation for rhea:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[kegg.reaction]\u001b[0m - AssertionError: The following 1044 reactions (67.84%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[seed.reaction]\u001b[0m - AssertionError: The following 574 reactions (37.30%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[metanetx.reaction]\u001b[0m - AssertionError: The following 223 reactions (14.49%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[bigg.reaction]\u001b[0m - AssertionError: The following 36 reactions (2.34%) lack annotation for bigg...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[reactome]\u001b[0m - AssertionError: The following 1355 reactions (88.04%) lack annotation for r...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[ec-code]\u001b[0m - AssertionError: The following 831 reactions (54.00%) lack annotation for ec...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[brenda]\u001b[0m - AssertionError: The following 1539 reactions (100.00%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[biocyc]\u001b[0m - AssertionError: The following 898 reactions (58.35%) lack annotation for bi...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[refseq]\u001b[0m - AssertionError: The following 805 genes (100.00%) lack annotation for refseq:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[uniprot]\u001b[0m - AssertionError: The following 805 genes (100.00%) lack annotation for uniprot:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ecogene]\u001b[0m - AssertionError: The following 805 genes (100.00%) lack annotation for ecogene:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[kegg.genes]\u001b[0m - AssertionError: The following 68 genes (8.45%) lack annotation for kegg.genes:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ncbigi]\u001b[0m - AssertionError: The following 805 genes (100.00%) lack annotation for ncbigi:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ncbigene]\u001b[0m - AssertionError: The following 68 genes (8.45%) lack annotation for ncbigene:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ncbiprotein]\u001b[0m - AssertionError: The following 24 genes (2.98%) lack annotation for ncbiprot...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ccds]\u001b[0m - AssertionError: The following 805 genes (100.00%) lack annotation for ccds:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[hprd]\u001b[0m - AssertionError: The following 805 genes (100.00%) lack annotation for hprd:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[asap]\u001b[0m - AssertionError: The following 805 genes (100.00%) lack annotation for asap:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[pubchem.compound]\u001b[0m - AssertionError: There are no metabolite annotations for the pubchem.compoun...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[kegg.compound]\u001b[0m - AssertionError: A total of 274 metabolite annotations (33.91%) do not match...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[seed.compound]\u001b[0m - AssertionError: A total of 40 metabolite annotations (4.15%) do not match the\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[inchi]\u001b[0m - AssertionError: There are no metabolite annotations for the inchi database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[hmdb]\u001b[0m - AssertionError: A total of 181 metabolite annotations (25.28%) do not match...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[reactome]\u001b[0m - AssertionError: A total of 411 metabolite annotations (88.01%) do not match...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[ec-code]\u001b[0m - AssertionError: A total of 4 reaction annotations (0.26%) do not match the\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[brenda]\u001b[0m - AssertionError: There are no reaction annotations for the brenda database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[refseq]\u001b[0m - AssertionError: There are no gene annotations for the refseq database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[uniprot]\u001b[0m - AssertionError: There are no gene annotations for the uniprot database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ecogene]\u001b[0m - AssertionError: There are no gene annotations for the ecogene database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ncbigi]\u001b[0m - AssertionError: There are no gene annotations for the ncbigi database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ncbigene]\u001b[0m - AssertionError: A total of 737 gene annotations (91.55%) do not match the\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ncbiprotein]\u001b[0m - AssertionError: A total of 1 gene annotations (0.12%) do not match the     ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ccds]\u001b[0m - AssertionError: There are no gene annotations for the ccds database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[hprd]\u001b[0m - AssertionError: There are no gene annotations for the hprd database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[asap]\u001b[0m - AssertionError: There are no gene annotations for the asap database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_gene_protein_reaction_rule_presence\u001b[0m - AssertionError: There are a total of 276 reactions (17.93%) without GPR:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_transport_reaction_gpr_presence\u001b[0m - AssertionError: There are a total of 110 transport reactions (24.72% of all\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_duplicate_metabolites_in_compartments\u001b[0m - TypeError: sequence item 0: expected str instance, tuple found\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_reactions_with_partially_identical_annotations\u001b[0m - AssertionError: Based on annotations there are 97 different groups of overl...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_duplicate_reactions\u001b[0m - TypeError: sequence item 0: expected str instance, tuple found\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_reactions_with_identical_genes\u001b[0m - AssertionError: Based only on equal genes there are 172 different groups of\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_medium_metabolites\u001b[0m - TypeError: unsupported operand type(s) for /: 'int' and 'list'\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_biomass_consistency[Growth]\u001b[0m - AssertionError: The component molar mass of the biomass reaction Growth sum...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_gam_in_biomass[Growth]\u001b[0m - AssertionError: No, Growth does not contain a term for growth-associated\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_fast_growth_default[Growth]\u001b[0m - AssertionError: Using the biomass reaction Growth and when the model is sim...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_essential_precursors_not_in_biomass[Growth]\u001b[0m - AssertionError: Growth lacks a total of 1 essential metabolites         (1....\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_stoichiometric_consistency\u001b[0m - AssertionError: This model's stoichiometry inconsistent\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_unconserved_metabolites\u001b[0m - AssertionError: This model contains 3 unconserved metabolites: h_e, h_p, h_c\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_inconsistent_min_stoichiometry\u001b[0m - TypeError: sequence item 0: expected str instance, list found\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_reaction_charge_balance\u001b[0m - AssertionError: A total of 28 (2.07%) reactions are charge unbalanced with at\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_reaction_mass_balance\u001b[0m - AssertionError: A total of 11 (0.81%) reactions are mass unbalanced with at...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_blocked_reactions\u001b[0m - AssertionError: daemonic processes are not allowed to have children\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_stoichiometrically_balanced_cycles\u001b[0m - AssertionError: daemonic processes are not allowed to have children\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_orphans\u001b[0m - AssertionError: A total of 1 (0.10%) metabolites are not produced by any re...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_deadends\u001b[0m - AssertionError: A total of 1 (0.10%) metabolites are not consumed by any re...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_metabolites_not_produced_with_open_bounds\u001b[0m - AssertionError: daemonic processes are not allowed to have children\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_metabolites_not_consumed_with_open_bounds\u001b[0m - AssertionError: daemonic processes are not allowed to have children\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_reactions_unbounded_flux_default_condition\u001b[0m - AssertionError: daemonic processes are not allowed to have children\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_metabolite_sbo_presence\u001b[0m - AssertionError: A total of 16 metabolites (1.54%) lack annotation with any ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_reaction_sbo_presence\u001b[0m - AssertionError: A total of 3 reactions (0.19%) lack annotation with any type\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_gene_sbo_presence\u001b[0m - AssertionError: A total of 25 genes (3.11%) lack annotation with any type of\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_metabolic_reaction_specific_sbo_presence\u001b[0m - AssertionError: A total of 604 metabolic reactions (66.59% of all purely\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_transport_reaction_specific_sbo_presence\u001b[0m - AssertionError: A total of 9 metabolic reactions (2.02% of all transport\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_metabolite_specific_sbo_presence\u001b[0m - AssertionError: A total of 16 transport reactions (1.54% of all metabolites)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_gene_specific_sbo_presence\u001b[0m - AssertionError: A total of 25 genes (3.11% of all genes) lack             a...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_exchange_specific_sbo_presence\u001b[0m - AssertionError: A total of 1 exchange reactions (0.55% of all exchange\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_sink_specific_sbo_presence\u001b[0m - AssertionError: A total of 0 genes (0.00% of all sink reactions) lack\n",
      "\u001b[31m============= \u001b[31m\u001b[1m82 failed\u001b[0m, \u001b[32m59 passed\u001b[0m, \u001b[33m6 skipped\u001b[0m, \u001b[33m1 warning\u001b[0m\u001b[31m in 44.99s\u001b[0m\u001b[31m =============\u001b[0m\n",
      "\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[32m.\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31mF\u001b[0m\u001b[31m                                             [ 86%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_essentiality.py \u001b[33ms\u001b[0m\u001b[31m [ 87%]\n",
      "\u001b[0m\u001b[31m                                                                         [ 87%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_growth.py \u001b[33ms\u001b[0m\u001b[31m [ 88%]\n",
      "\u001b[0m\u001b[31m                                                                         [ 88%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_matrix.py \u001b[32m.\u001b[0m\u001b[31m [ 89%]\n",
      "\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[31m                                                                      [ 91%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbml.py \u001b[32m.\u001b[0m\u001b[31m [ 91%]\n",
      "\u001b[0m\u001b[32m.\u001b[0m\u001b[31m                                                                        [ 92%]\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py \u001b[32m.\u001b[0m\u001b[31m [ 93%]\n",
      "\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[33ms\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[32m.\u001b[0m\u001b[33ms\u001b[0m\u001b[33ms\u001b[0m\u001b[31mF\u001b[0m\u001b[31m                                                               [100%]\u001b[0m\n",
      "\n",
      "\u001b[33m=============================== warnings summary ===============================\u001b[0m\n",
      "../../../anaconda3/lib/python3.11/site-packages/_pytest/config/__init__.py:1204\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/_pytest/config/__init__.py:1204: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: anyio\n",
      "    self._mark_plugins_for_rewrite(hook)\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::test_biomass_consistency[CG_biomass_cgl_ATCC13032]\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 26.227799999999995 is not an integer (in formula C26.227799999999995H47.309599999999996O21.0818)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::test_biomass_consistency[CG_biomass_cgl_ATCC13032]\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 47.309599999999996 is not an integer (in formula C26.227799999999995H47.309599999999996O21.0818)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::test_biomass_consistency[CG_biomass_cgl_ATCC13032]\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 21.0818 is not an integer (in formula C26.227799999999995H47.309599999999996O21.0818)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::test_biomass_consistency[CG_biomass_cgl_ATCC13032]\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:191: UserWarning: The element 'R' does not appear in the periodic table\n",
      "    warn(f\"The element {e} does not appear in the periodic table\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 29.018 is not an integer (in formula C29.018H52.94199999999999O15P)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 52.94199999999999 is not an integer (in formula C29.018H52.94199999999999O15P)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 41.018 is not an integer (in formula C41.018H74.942O29P2)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 74.942 is not an integer (in formula C41.018H74.942O29P2)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 46.2 is not an integer (in formula C46.2H80.208N3O15P2)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 80.208 is not an integer (in formula C46.2H80.208N3O15P2)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 17.018 is not an integer (in formula C17.018H30.941999999999997O1)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 30.941999999999997 is not an integer (in formula C17.018H30.941999999999997O1)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 40.2 is not an integer (in formula C40.2H75.208O13P2)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 75.208 is not an integer (in formula C40.2H75.208O13P2)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 20.1 is not an integer (in formula C20.1H38.104O7P)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 38.104 is not an integer (in formula C20.1H38.104O7P)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 1.809384 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.452346 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 1.514453 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.542815 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.002624 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.073913 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 20.35556 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.904692 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.072104 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.033655 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.035554 is not an integer (in formula O1.809384S0.452346Fe1.514453Ca0.542815Co0.002624Cu0.073913K20.35556Mg0.904692Mn0.072104Ni0.033655Zn0.035554Cl0.542815)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 64.598238154 is not an integer (in formula C64.598238154H117.50071172599999O7.357917P0.267926)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 117.50071172599999 is not an integer (in formula C64.598238154H117.50071172599999O7.357917P0.267926)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 7.357917 is not an integer (in formula C64.598238154H117.50071172599999O7.357917P0.267926)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.267926 is not an integer (in formula C64.598238154H117.50071172599999O7.357917P0.267926)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 64.087066 is not an integer (in formula C64.087066H109.98235400000002N13.276022O50.168144P3.2394)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 109.98235400000002 is not an integer (in formula C64.087066H109.98235400000002N13.276022O50.168144P3.2394)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 13.276022 is not an integer (in formula C64.087066H109.98235400000002N13.276022O50.168144P3.2394)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 50.168144 is not an integer (in formula C64.087066H109.98235400000002N13.276022O50.168144P3.2394)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 3.2394 is not an integer (in formula C64.087066H109.98235400000002N13.276022O50.168144P3.2394)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 52.881848160000004 is not an integer (in formula C52.881848160000004H98.21897442240001N0.326586O16.9293528P2.563819800000001)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 98.21897442240001 is not an integer (in formula C52.881848160000004H98.21897442240001N0.326586O16.9293528P2.563819800000001)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.326586 is not an integer (in formula C52.881848160000004H98.21897442240001N0.326586O16.9293528P2.563819800000001)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 16.9293528 is not an integer (in formula C52.881848160000004H98.21897442240001N0.326586O16.9293528P2.563819800000001)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 2.563819800000001 is not an integer (in formula C52.881848160000004H98.21897442240001N0.326586O16.9293528P2.563819800000001)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 28.60628 is not an integer (in formula C28.60628H33.27465N11.5971O22.976729999999996P2.99579)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 33.27465 is not an integer (in formula C28.60628H33.27465N11.5971O22.976729999999996P2.99579)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 11.5971 is not an integer (in formula C28.60628H33.27465N11.5971O22.976729999999996P2.99579)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 22.976729999999996 is not an integer (in formula C28.60628H33.27465N11.5971O22.976729999999996P2.99579)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 2.99579 is not an integer (in formula C28.60628H33.27465N11.5971O22.976729999999996P2.99579)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 19.222443 is not an integer (in formula C19.222443H38.476530999999966N5.458296999999999O10.650917999999953P-0.000000000000004S0.108442)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 38.476530999999966 is not an integer (in formula C19.222443H38.476530999999966N5.458296999999999O10.650917999999953P-0.000000000000004S0.108442)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 5.458296999999999 is not an integer (in formula C19.222443H38.476530999999966N5.458296999999999O10.650917999999953P-0.000000000000004S0.108442)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 10.650917999999953 is not an integer (in formula C19.222443H38.476530999999966N5.458296999999999O10.650917999999953P-0.000000000000004S0.108442)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.108442 is not an integer (in formula C19.222443H38.476530999999966N5.458296999999999O10.650917999999953P-0.000000000000004S0.108442)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 31.685756 is not an integer (in formula C31.685756H40.884037000000006N9.559588O20.989518P3.134647S0.355614)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 40.884037000000006 is not an integer (in formula C31.685756H40.884037000000006N9.559588O20.989518P3.134647S0.355614)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 9.559588 is not an integer (in formula C31.685756H40.884037000000006N9.559588O20.989518P3.134647S0.355614)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 20.989518 is not an integer (in formula C31.685756H40.884037000000006N9.559588O20.989518P3.134647S0.355614)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 3.134647 is not an integer (in formula C31.685756H40.884037000000006N9.559588O20.989518P3.134647S0.355614)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::test_reaction_mass_balance\n",
      "  /Users/macintosh/anaconda3/lib/python3.11/site-packages/cobra/core/metabolite.py:146: UserWarning: 0.355614 is not an integer (in formula C31.685756H40.884037000000006N9.559588O20.989518P3.134647S0.355614)\n",
      "    warn(f\"{count} is not an integer (in formula {self.formula})\")\n",
      "\n",
      "-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html\n",
      "\u001b[36m\u001b[1m=========================== short test summary info ============================\u001b[0m\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_presence\u001b[0m - AssertionError: A total of 949 metabolites (100.00%) lack any form of annot...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_presence\u001b[0m - AssertionError: A total of 1207 reactions (100.00%) lack any form of annota...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_presence\u001b[0m - AssertionError: A total of 773 genes (100.00%) lack any form of         ann...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[pubchem.compound]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[kegg.compound]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[seed.compound]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[inchikey]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[inchi]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[chebi]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[hmdb]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[reactome]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[metanetx.chemical]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[bigg.metabolite]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_overview[biocyc]\u001b[0m - AssertionError: The following 949 metabolites (100.00%) lack annotation for...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[rhea]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[kegg.reaction]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[seed.reaction]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[metanetx.reaction]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[bigg.reaction]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[reactome]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[ec-code]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[brenda]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_overview[biocyc]\u001b[0m - AssertionError: The following 1207 reactions (100.00%) lack annotation for ...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[refseq]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for refseq:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[uniprot]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for uniprot:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ecogene]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for ecogene:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[kegg.genes]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for kegg....\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ncbigi]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for ncbigi:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ncbigene]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for ncbig...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ncbiprotein]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for ncbip...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[ccds]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for ccds:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[hprd]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for hprd:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_overview[asap]\u001b[0m - AssertionError: The following 773 genes (100.00%) lack annotation for asap:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[pubchem.compound]\u001b[0m - AssertionError: There are no metabolite annotations for the pubchem.compoun...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[kegg.compound]\u001b[0m - AssertionError: There are no metabolite annotations for the kegg.compound d...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[seed.compound]\u001b[0m - AssertionError: There are no metabolite annotations for the seed.compound d...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[inchikey]\u001b[0m - AssertionError: There are no metabolite annotations for the inchikey database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[inchi]\u001b[0m - AssertionError: There are no metabolite annotations for the inchi database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[chebi]\u001b[0m - AssertionError: There are no metabolite annotations for the chebi database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[hmdb]\u001b[0m - AssertionError: There are no metabolite annotations for the hmdb database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[reactome]\u001b[0m - AssertionError: There are no metabolite annotations for the reactome database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[metanetx.chemical]\u001b[0m - AssertionError: There are no metabolite annotations for the metanetx.chemical\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[bigg.metabolite]\u001b[0m - AssertionError: There are no metabolite annotations for the bigg.metabolite...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_metabolite_annotation_wrong_ids[biocyc]\u001b[0m - AssertionError: There are no metabolite annotations for the biocyc database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[rhea]\u001b[0m - AssertionError: There are no reaction annotations for the rhea database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[kegg.reaction]\u001b[0m - AssertionError: There are no reaction annotations for the kegg.reaction dat...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[seed.reaction]\u001b[0m - AssertionError: There are no reaction annotations for the seed.reaction dat...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[metanetx.reaction]\u001b[0m - AssertionError: There are no reaction annotations for the metanetx.reaction...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[bigg.reaction]\u001b[0m - AssertionError: There are no reaction annotations for the bigg.reaction dat...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[reactome]\u001b[0m - AssertionError: There are no reaction annotations for the reactome database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[ec-code]\u001b[0m - AssertionError: There are no reaction annotations for the ec-code database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[brenda]\u001b[0m - AssertionError: There are no reaction annotations for the brenda database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_reaction_annotation_wrong_ids[biocyc]\u001b[0m - AssertionError: There are no reaction annotations for the biocyc database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[refseq]\u001b[0m - AssertionError: There are no gene annotations for the refseq database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[uniprot]\u001b[0m - AssertionError: There are no gene annotations for the uniprot database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ecogene]\u001b[0m - AssertionError: There are no gene annotations for the ecogene database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[kegg.genes]\u001b[0m - AssertionError: There are no gene annotations for the kegg.genes database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ncbigi]\u001b[0m - AssertionError: There are no gene annotations for the ncbigi database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ncbigene]\u001b[0m - AssertionError: There are no gene annotations for the ncbigene database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ncbiprotein]\u001b[0m - AssertionError: There are no gene annotations for the ncbiprotein database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[ccds]\u001b[0m - AssertionError: There are no gene annotations for the ccds database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[hprd]\u001b[0m - AssertionError: There are no gene annotations for the hprd database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_annotation.py::\u001b[1mtest_gene_product_annotation_wrong_ids[asap]\u001b[0m - AssertionError: There are no gene annotations for the asap database.\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_gene_protein_reaction_rule_presence\u001b[0m - AssertionError: There are a total of 247 reactions (20.46%) without GPR:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_ngam_presence\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_compartments_presence\u001b[0m - AssertionError: A total of 1 compartments are defined in the model:\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_protein_complex_presence\u001b[0m -   File \"<unknown>\", line 1\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_transport_reactions\u001b[0m - AssertionError: A total of 0 (0.00%) transport reactions are defined in the\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_constrained_transport_reactions\u001b[0m - ZeroDivisionError: division by zero\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_transport_reaction_gpr_presence\u001b[0m - ZeroDivisionError: division by zero\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_unique_metabolites\u001b[0m - AssertionError: Not counting the same entities in other compartments, there...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_reactions_with_identical_genes\u001b[0m - AssertionError: Based only on equal genes there are 102 different groups of\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_basic.py::\u001b[1mtest_find_medium_metabolites\u001b[0m - TypeError: unsupported operand type(s) for /: 'int' and 'list'\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_biomass_consistency[CG_biomass_cgl_ATCC13032]\u001b[0m - TypeError: '<' not supported between instances of 'float' and 'NoneType'\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_biomass_precursors_default_production[CG_biomass_cgl_ATCC13032]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_biomass_precursors_open_production[CG_biomass_cgl_ATCC13032]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_gam_in_biomass[CG_biomass_cgl_ATCC13032]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_fast_growth_default[CG_biomass_cgl_ATCC13032]\u001b[0m - AssertionError: Using the biomass reaction CG_biomass_cgl_ATCC13032 and whe...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_direct_metabolites_in_biomass[CG_biomass_cgl_ATCC13032]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_biomass.py::\u001b[1mtest_essential_precursors_not_in_biomass[CG_biomass_cgl_ATCC13032]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_stoichiometric_consistency\u001b[0m - AssertionError: This model's stoichiometry inconsistent\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_unconserved_metabolites\u001b[0m - AssertionError: This model contains 5 unconserved metabolites: h2_c, h_c, h...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_inconsistent_min_stoichiometry\u001b[0m - TypeError: sequence item 0: expected str instance, list found\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM3]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM63]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM51]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM121]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM423]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM6]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM10]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM38]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM208]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM191]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM223]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM7517]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM12233]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM558]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM21]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_detect_energy_generating_cycles[MNXM89557]\u001b[0m - ValueError: not enough values to unpack (expected 2, got 1)\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_reaction_mass_balance\u001b[0m - AssertionError: A total of 32 (3.07%) reactions are mass unbalanced with at...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_blocked_reactions\u001b[0m - AssertionError: daemonic processes are not allowed to have children\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_stoichiometrically_balanced_cycles\u001b[0m - AssertionError: daemonic processes are not allowed to have children\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_deadends\u001b[0m - AssertionError: A total of 0 (0.00%) metabolites are not consumed by any re...\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_metabolites_not_produced_with_open_bounds\u001b[0m - AssertionError: daemonic processes are not allowed to have children\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_metabolites_not_consumed_with_open_bounds\u001b[0m - AssertionError: daemonic processes are not allowed to have children\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_consistency.py::\u001b[1mtest_find_reactions_unbounded_flux_default_condition\u001b[0m - AssertionError: daemonic processes are not allowed to have children\n",
      "\u001b[31mFAILED\u001b[0m ../../../anaconda3/lib/python3.11/site-packages/memote/suite/tests/test_sbo.py::\u001b[1mtest_biomass_specific_sbo_presence\u001b[0m - AssertionError: A total of 1 biomass reactions (100.00% of all biomass reac...\n",
      "\u001b[31m====== \u001b[31m\u001b[1m107 failed\u001b[0m, \u001b[32m34 passed\u001b[0m, \u001b[33m6 skipped\u001b[0m, \u001b[33m61 warnings\u001b[0m\u001b[31m in 68.20s (0:01:08)\u001b[0m\u001b[31m =======\u001b[0m\n",
      "Writing diff report to 'index.html'.\n"
     ]
    }
   ],
   "source": [
    "!memote report diff /Users/macintosh/Downloads/iCGB21FR.xml /Users/macintosh/Downloads/iCW773.xml"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
